cmd/peer-rpc-client-target.go

```diff
@@ -31,11 +31,41 @@ func (target *PeerRPCClientTarget) ID() event.TargetID {
 	return target.id
 }
 
-// Send - sends event to remote peer by making RPC call.
-func (target *PeerRPCClientTarget) Send(eventData event.Event) error {
+// Save - sends event to remote peer by making RPC call.
+func (target *PeerRPCClientTarget) Save(eventData event.Event) error {
 	return target.rpcClient.SendEvent(target.bucketName, target.id, target.remoteTargetID, eventData)
 }
 
+// IsConnected - To be implemented.
+func (target *PeerRPCClientTarget) IsConnected() bool {
+	return true
+}
+
+// GetEvent - To be implemented.
+func (target *PeerRPCClientTarget) GetEvent(eventKey string) (event.Event, error) {
+	var event event.Event
+	return event, nil
+}
+
+// DelEvent - To be implemented.
+func (target *PeerRPCClientTarget) DelEvent(eventKey string) {
+}
+
+// GetEventKeyCh - To be implemented.
+func (target *PeerRPCClientTarget) GetEventKeyCh() chan string {
+	return nil
+}
+
+// GetDoneCh - To be implemented.
+func (target *PeerRPCClientTarget) GetDoneCh() chan struct{} {
+	return nil
+}
+
+// Send - sends event to Client target.
+func (target *PeerRPCClientTarget) Send(eventData event.Event) error {
+	return nil
+}
+
```

Comment:
I can think of below;

1. event persistent store is singleton global service available for all target and event sender service is available for each target.
2. At `Target`, event persistent store is used to store event when `Save()` is called and no other place the service is required.
3. A `Target` has event sender service that reads single event from event persistent store and calls `target.Send()` to push it to the target service.
4. Upon successful `target.Send()` the event is removed from event persistent store; else `target.Send()` is attempted infinitely till success.
4. event persistent store may have built-in in-memory store if disk it not available to persist events.

With above in mind; A `Target` needs below methods
```
type Target interface {
    Send(event) error
    Save(event) error
}
```

Rest are unnecessary and complicating.

Reply:
Yes. But, the SendEvents function here https://github.com/minio/minio/pull/7343/files#diff-9c849d8e4afbc5ffa2e6b204b493a386R158 is generically written. It needs few functions implemented by all the targets. Otherwise, this code has to be duplicated in all the targets.

Each function here serves its own purpose. For eg, 

GetEvent - This reads an event key from the channel.
IsConnected - Polls for the conn status.
Send - sends the event to the target.
DelEvent - Deletes the event from the store.

I though these functions are generic enough to serve in interface.

Reply:
Resolved this through slack. closing this as resolved.